### PR TITLE
fix divide by zero bug; add distance logging when speed == 0

### PR DESCRIPTION
**Pull request by leanhdaovn**
- fix divide by zero bug
- add distance logging when speed == 0

### CODE REVIEW

File: user_web_catchable [hunk 1]
```diff
@@ -0,0 +1 @@
+{"pokemon_id": 98, "longitude": 106.68932778146748, "expiration_timestamp_ms": 1470605381948, "latitude": 10.770729135498259, "spawn_point_id": "31752f3cdf7", "encounter_id": 7175942365001798541}
```
Comment (TheSavior) [inline user_web_catchable]: Did you mean to modify this file?

File: pokemongo_bot/cell_workers/follow_spiral.py [hunk 2]
```diff
@@ -66,9 +66,19 @@ def _generate_spiral(starting_lat, starting_lng, step_size, step_limit):
...
 
     def work(self):
+        last_lat = self.bot.api._position_lat
+        last_lng = self.bot.api._position_lng
+
         point = self.points[self.ptr]
         self.cnt += 1
 
+        dist = distance(
+            self.bot.api._position_lat,
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/follow_spiral.py]: Why doesn't this use last_lat?

**Comment by leanhdaovn** [comment]
@TheSavior thanks for commenting. I just fixed those changes.

File: pokemongo_bot/cell_workers/follow_spiral.py [hunk 3]
```diff
@@ -77,19 +87,12 @@ def work(self):
...
-                point['lat'],
-                point['lng']
-            )
-
             if self.cnt == 1:
                 self.emit_event(
                     'position_update',
                     formatted="Walking from {last_position} to {current_position} ({distance} {distance_unit})",
                     data={
-                        'last_position': self.bot.position,
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/follow_spiral.py]: Why did this one need to change?

Comment (leanhdaovn) [inline pokemongo_bot/cell_workers/follow_spiral.py]: just to make it consistent with other places' usage

**Comment by TheSavior** [comment]
:+1:

**Comment by k4n30** [comment]
@TheSavior & @leanhdaovn This merge is causing #3920

**Comment by leanhdaovn** [comment]
@k4n30 
The merge caused a wrong message (mistake from copy-paste), not a logic bug. Let me fix it.

**Comment by leanhdaovn** [comment]
New PR: #3935

**Pull request merged**